apps/wallet/src/ui/app/pages/swap/SwapPageForm.tsx

```diff
@@ -0,0 +1,460 @@
+
+	const activeCoinType = searchParams.get('type');
+
+	const { isLoading, tokenBalance, coinMetadata } = useCoinTypeData(activeCoinType);
+
+	const { data: coinBalances } = useSuiClientQuery(
+		'getAllBalances',
+		{ owner: activeAccountAddress! },
+		{
+			enabled: !!activeAccountAddress,
+			staleTime,
+			refetchInterval,
+			select: filterAndSortTokenBalances,
+		},
+	);
+
+	const { recognized } = useSortedCoinsByCategories(coinBalances ?? []);
+
+	const formattedTokenBalance = tokenBalance.replace(/,/g, '');
+	const symbol = coinMetadata.data?.symbol ?? '';
+
+	const coinDecimals = coinMetadata.data?.decimals ?? 0;
+	const balanceInMist = new BigNumber(tokenBalance || '0')
+		.times(MIST_PER_SUI.toString())
+		.toString();
+
+	const validationSchema = useMemo(() => {
+		return validate(BigInt(balanceInMist), symbol, coinDecimals);
+	}, [balanceInMist, coinDecimals, symbol]);
+
+	const renderButtonToCoinsList = useMemo(() => {
+		return (
+			recognized.length > 1 &&
+			recognized.some((coin) => allowedSwapCoinsList.includes(coin.coinType))
+		);
+	}, [recognized]);
+
+	return (
+		<Overlay showModal title="Swap" closeOverlay={() => navigate('/')}>
+			<div className="flex flex-col w-full h-full">
+				<Loading loading={isLoading}>
+					<Formik
+						initialValues={initialValues}
+						onSubmit={() => {}}
+						validationSchema={validationSchema}
+						enableReinitialize
+						validateOnMount
+						validateOnChange
+					>
+						{({ isValid, isSubmitting, setFieldValue, values, submitForm, validateField }) => {
+							const newIsPayAll = !!values.amount && values.amount === tokenBalance;
+
+							if (values.isPayAll !== newIsPayAll) {
+								setFieldValue('isPayAll', newIsPayAll);
+							}
+
+							return (
+								<BottomMenuLayout>
+									<Content>
+										<Form autoComplete="off" noValidate>
+											<div
+												className={clsx(
+													'flex flex-col border border-hero-darkest/20 rounded-xl pt-5 pb-6 px-5 gap-4 border-solid',
+													isValid && 'bg-gradients-graph-cards',
+												)}
+											>
+												{activeCoinType && (
+													<AssetData
+														disabled={!renderButtonToCoinsList}
+														tokenBalance={tokenBalance}
+														coinType={activeCoinType}
+														symbol={symbol}
+														to="/swap/base-assets"
+													/>
+												)}
+												<InputWithAction
+													type="numberInput"
+													name="amount"
+													placeholder="0.00"
+													prefix={values.isPayAll ? '~ ' : ''}
+													actionText="Max"
+													actionDisabled={values.isPayAll}
+													suffix={` ${symbol}`}
+													actionType="button"
+													allowNegative={false}
+													decimals
+													rounded="lg"
+													dark
+													onActionClicked={async () => {
+														// using await to make sure the value is set before the validation
```

Comment:
is this necessary? it seems by default [setFieldValue](https://formik.org/docs/api/formik#setfieldvalue-field-string-value-any-shouldvalidate-boolean--promisevoid--formikerrors) does the validation

Reply:
Updated to use react hook form